blog/_posts/2021-07-31-Mitchell-Johnson_Jamie-Schmidt.md

```diff
@@ -0,0 +1,118 @@
+---
+layout: topic
+title: "Development of MABE2"
+date: 2021-07-31
+author: Mitchell Johnson & Jamie Schmidt
+---
+
+### Development of MABE2
+
+## Introduction
+-what is MABE2
+For our project, we focused our work mostly on 
+-improving the end user experience through the development of DynamicOrg.
+-adding support for brains and genomes, both of which have base classes
+
+To quote the [Github for MABE2](https://github.com/mercere99/MABE2), MABE is a software framework deigned to easily build and customize software for evolutionary computation or artificial life.  The resulting systems should be useful for studying evolutionary dynamics, solving complex problems, comparing evolving systems, or exploring the open-ended power of evolution.  
+
+MABE, accessible [here](https://github.com/Hintzelab/MABE/wiki), stands for Modular Agent-Based Evolution and was one of the first true digital evolution frameworks.  It was structured in this fashion:
+
+![MABE structure](https://github.com/Hintzelab/MABE/wiki/images/MABE_Overview.png)
+
+and, though powerful, could of course be improved.  The creation of the 2.0 version of MABE, called MABE2, was the focus of our project.  Our main focus was on refining the underlying architecture of MABE to be more end user-friendly, particularly in designing custom experiments.  The culmination of our efforts was in the creation of DynamicOrg, an Organism which was near-completely abstracted to allow for any combination of Genomes and Brains, as assigned through the config file. 
+
+## Genomes / author: jamie
+Genomes in MABE2 are just repositories of mutatable information.  The most basic kinds are vectors of some type, and (unlike in MABE), they aren't hidden inside Organisms which manage everything about them, but they are actively added to them and manage themselves.  This means that to mutate a Genome, any Genome, the Organism just tells the Genome to mutate itself and the Genome does.  No longer will the Organism run all the mutation functions itself, now it just delegates.  This, critically, means that Organisms now no longer need to know what kinds of Genomes they have at all.
```

Comment:
Dig deeper into what the motivations/advantages are. What types of organisms or what types of savings of developer time are possible?

Reply:
done